File: protocol/src/test/typescript/zfoo/packet/EmptyObject.ts


class EmptyObject {

    

    static PROTOCOL_ID: number = 0;

    protocolId(): number {
        return EmptyObject.PROTOCOL_ID;
    }

    static write(buffer: any, packet: EmptyObject | null) {
        if (packet === null) {
            buffer.writeInt(0);
            return;
        }
        buffer.writeInt(-1);
    }

    static read(buffer: any): EmptyObject | null {
        const length = buffer.readInt();
        if (length === 0) {
            return null;
        }
        const beforeReadIndex = buffer.getReadOffset();
        const packet = new EmptyObject();
        
        if (length > 0) {
            buffer.setReadOffset(beforeReadIndex + length);
        }
        return packet;
    }
}

export default EmptyObject;

File: protocol/src/test/go/zfoogo/EmptyObject.go
package protocol

type EmptyObject struct {
	
}

func (protocol EmptyObject) ProtocolId() int16 {
	return 0
}

func (protocol EmptyObject) write(buffer *ByteBuffer, packet any) {
	if packet == nil {
	    buffer.WriteInt(0)
		return
	}
	buffer.WriteInt(-1)
}

func (protocol EmptyObject) read(buffer *ByteBuffer) any {
	var packet = new(EmptyObject)
	var length = buffer.ReadInt()
	if length == 0 {
		return packet
	}
	var beforeReadIndex = buffer.ReadOffset()
	if length > 0 {
        buffer.SetReadOffset(beforeReadIndex + length)
    }
	return packet
}
